### PR TITLE
Gemfile - update engine cart stanza for v1.1.0

### PR DESCRIPTION
**Pull request by darrenleeweber**
Related to https://github.com/cbeer/engine_cart/issues/79 (does not fix it in engine_cart)

@projecthydra-labs/hyrax-code-reviewers

@mjgiarlo - I just discovered that this could undo a prior hack on the engine-cart stanza:
  - https://github.com/projecthydra-labs/hyrax/commit/8071e331a5d522c42dc104bbfaa7f64b7cb422c8
  - Is there another way to achieve that without hacking the engine-cart stanza?
    - attempted to do so in https://github.com/projecthydra-labs/hyrax/pull/1010/commits/fefd4901a85d4be13ec2ff2cb5a418a1d66d6b99
  - review commentary resolved to ignore overriding this gem version for rails 4.2 because hyrax needs rails 5.x now anyway, so the commit ^^ was removed.

### CODE REVIEW

**Comment by darrenleeweber** [comment]
@cbeer - add rubocop checks on engine_cart so it's stanza passes rubocop checks.

```
Gemfile:41:5: C: Bundler/OrderedGems: Gems should be sorted in an alphabetical order within their section of the Gemfile. Gem coffee-rails should appear before sass-rails.

    gem 'coffee-rails', '~> 4.1.0'

    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: Gemfile [hunk 1]
```diff
@@ -8,10 +8,15 @@ group :development, :test do
   gem 'pry' unless ENV['CI']
   gem 'pry-byebug' unless ENV['CI']
   gem "simplecov", require: false
+  # Override the coffee-rails version in engine_cart for rails 4.2
+  case ENV['RAILS_VERSION']
+  when /^4.2/
+    gem 'coffee-rails', '~> 4.2.0'
```
Comment (jcoyne) [inline Gemfile]: This seems like it conflicts with line 44.  Can you say more about why we need both?

Comment (darrenleeweber) [inline Gemfile]: The line-44 code is maintained by engine-cart, it should not be modified by anything but engine cart and we should assume that engine-cart will not respect any changes in the block it maintains.  To override anything in that block, it must be declared separately.  So, in this PR, to preserve the intention of a prior commit, this block was added in the last commit.

Comment (darrenleeweber) [inline Gemfile]: A separate issue is whether this override is actually needed at all any more, so I did not intend to address that in this PR.

Comment (cbeer) [inline Gemfile]: > The line-44 code is maintained by engine-cart, it should not be modified by anything but engine cart

I wouldn't go that far -- instead, I'd say: by modifying it, the onus is on the project maintainers to ensure it is maintained the way they expect.

I thought having duplicate gem declarations (particularly conflicting ones) makes bundler unhappy?

Comment (darrenleeweber) [inline Gemfile]: Good question - I have not tested what happens when the rails version hits this code.  If hyrax is no longer compatible with rails 4.x, then I should just remove the last commit and leave that stanza as engine-cart creates it (i.e. discard the override entirely).

Comment (cbeer) [inline Gemfile]: 👍

Comment (darrenleeweber) [inline Gemfile]: Removed the last commit to override anything on rails 4.2

**Pull request merged**